### ISSUE TITLE
Formatting option to remove trailing comma?

### ISSUE THREAD

**Issue by matthewmueller** [tailscale/hujson#36]
Thanks for creating this library! 

I know this goes against the `gofmt` tradition of no options, but it'd be nice if we could piggyback off VSCode's already supported file extension `.jsonc`.

Currently when you run `value.Format()`, it adds trailing commas by default. In VSCode, written as a `jsonc` file, this leads to warnings:

![CleanShot 2024-10-27 at 16 39 24](https://github.com/user-attachments/assets/86f69e42-acb1-46ef-b726-da26911b5c51)

I saw the comment in the Readme about defining a new `.hujson` extension, but I'd prefer not to have to ask users to do this. 

Some ideas:

- Would you be open to providing a list of options to the Format function? So we could do something like:
   ```go
   out, err := hujson.Format(hudata, hujson.Trailing(false))
   ```

- Alternatively, changing the `hujson.Format(hudata)` function to remove trailing commas would work too.

Oh and happy to contribute the change if we land on a solution!

**Comment by bradfitz** [comment]
I'd say this is @dsnet's decision, but he's on leave for a bit so might not reply soon.

**Comment by matthewmueller** [comment]
One other thing the default VSCode formatter does it turn tabs into two spaces, so ideally that would be an option too 😅 

I made these changes in [a fork](https://github.com/matthewmueller/hujson/commits/master/) that I'd be happy to PR or adjust based on feedback.

I do think two spaces is a bit more common in the Javascript community, e.g. NPM's package.json uses spaces: https://github.com/npm/npm/issues/4718, but I can definitely understand not wanting to introduce all these options. 

I do want to share that hujson, especially with its json patching supports is the only solution I'm aware of that allows you to work natively with Go structs while preserving comments! If you pair hujson with https://github.com/snorwin/jsonpatch, you essentially can get:

```go
schema, err := schemas.ReadFile("path/to/file")

// schemas is just an unmarshaled struct!
schema.Hosts = append(schema.hosts, schemas.Host{...})

// Read original schema from disk, patch it with new schema, write back to disk
schemas.WriteFile("path/to/file", schema)
```

Thanks for this library!

**Comment by matthewmueller** [comment]
Found another difference. This library does Go-style field alignment, while the default VSCode jsonc formatter (and Prettier) don't have this behavior.

At this point, I think this is basically a separate library. For now I've forked the library over at: https://github.com/matthewmueller/jsonc. I'd still love to upstream it somehow, perhaps allowing users to swap out the formatter, but either way, thanks for the great library!
